### ISSUE TITLE
WARNING in Circular dependency detected

### ISSUE THREAD

**Issue by pairmix** [typestack/class-transformer#230]
Hi there,
When i compile my ionic project with angular cli:
````
> ng build --prod
````
I get following warnings:
````
WARNING in Circular dependency detected:
src\app\entities\Membre.ts -> src\app\entities\MembreLocalisation.ts -> src\app\entities\Membre.ts

WARNING in Circular dependency detected:
src\app\entities\MembreLocalisation.ts -> src\app\entities\Membre.ts -> src\app\entities\MembreLocalisation.ts
````

Membre.ts:
````
import {Type} from "class-transformer";
import {MembreLocalisation} from "./MembreLocalisation";

export class Membre
{   
    id: number;
    email: string;

    @Type(() => MembreLocalisation) localisations: MembreLocalisation[];
}
````

MembreLocalisation.ts:
````
import {Type} from "class-transformer";
import {Membre} from "./Membre";

export class MembreLocalisation
{
    id: number;
    adresseip: string;
    countryname: string;
    
    @Type(() => Membre) membre: Membre;
}
````

How can i correct this ?

**Comment by jmesa-sistel** [comment]
@pairmix 
It is not a problem of the library, you are asking how to fix a circular dependency. The easy way here is have the 2 classes in the same file, the other option is use a barrel (I do not like barrels), a barrel is like a index.ts where you export both classes and you use this index.ts in the imports instead the file which contains the class.

Barrel solution: (as I said I do not like it)
index.ts:

```
export * from "./Membre";
export * from "./MembreLocalisation";

```
Membre.ts:

```
import {Type} from "class-transformer";
import {MembreLocalisation} from "./index";

export class Membre
{   
    id: number;
    email: string;

    @Type(() => MembreLocalisation) localisations: MembreLocalisation[];
}
```
MembreLocalisation.ts:

```
import {Type} from "class-transformer";
import {Membre} from "./index";

export class MembreLocalisation
{
    id: number;
    adresseip: string;
    countryname: string;
    
    @Type(() => Membre) membre: Membre;
}
```

**Comment by alexpls** [comment]
@pairmix did @jmesa-sistel's answer end up resolving your issue? If yes, would you mind closing this?

**Comment by NoNameProvided** [comment]
Will be solved by #380.
